### PR TITLE
QobjEvo no longer require cython for string coefficient

### PR DESCRIPTION
**Pull request by Ericgig**
Added the option to import string to python instead of compiling to cython for QobjEvo.
This is default if no cython, or set manually if cython available. `QobjEvoInstance.use_cython = False`
Added test and updated docstring.
Work in parallel.
Did not do any benchmark yet.

### CODE REVIEW

**Comment by ajgpitch** [comment]
Nice work Eric. I had a quick look at the code.
Does this PR mean that string type td will be used to construct Python functions when Cython not available? Just checking that I understand.

**Comment by Ericgig** [comment]
Yes, string will be turned to python function when cython is not available.
It is possible to use python even if cython is available if preferred.

**Comment by nathanshammah** [comment]
Is this merge-ready?

**Comment by Ericgig** [comment]
Wait 1 more commit so I check the test skipped without cython.

**Comment by Ericgig** [comment]
@nathanshammah 
That was a lot more than 1 commit but it work better.
- I added a test suite without cython. It only skip brmesolve time dependent tests.
- Error is raised when trying to use brmesolve without cython, but not at qutip import.

I tested without cython with both conda and pip. It work with both when done properly but conda is not as stable: uninstalling cython can change numpy, scipy, etc.'s versions which break qutip. So I am not sure if it will work properly we downloaded as binaries from conda. But this error should already be present for conda installation.

File: qutip/qobjevo_codegen.py [hunk 1]
```diff
@@ -38,12 +38,48 @@
...
+
+
+def _try_remove(filename):
+    try:
+        os.remove(filename)
+    except Exception:
+        pass
+
+
+def import_str(code, basefilename, obj_name, cythonfile=False):
```
Comment (nathanshammah) [inline qutip/qobjevo_codegen.py]: Can you add docstrings?

Comment (Ericgig) [reply qutip/qobjevo_codegen.py]: I made a few new functions to make codeclimat happy: small simple function, little repeated code.

**Comment by nathanshammah** [comment]
Tests still run fine, also against the current master, cc @tehruhn.

**Pull request merged**